types/validator_set.go

```diff
@@ -232,23 +232,23 @@ func (vals *ValidatorSet) HasAddress(address []byte) bool {
 	return idx < len(vals.Validators) && bytes.Equal(vals.Validators[idx].Address, address)
 }
 
-// GetByAddress returns an index of the validator with address and validator
-// itself if found. Otherwise, -1 and nil are returned.
-func (vals *ValidatorSet) GetByAddress(address []byte) (index int32, val *Validator) {
+// GetByAddress returns an index of the validator with address and validator if found.
+// Otherwise, 0, nil and a error are returned.
+func (vals *ValidatorSet) GetByAddress(address []byte) (index uint32, val *Validator, err error) {
```

Comment:
What's the benefit of adding error? aren't -1 and nil enough? I am struggling to see how adding an error here helps.

Reply:
can't return negative with uint32, I can check if its a -1 and convert otherwise. I am not the biggest fan of using -1 when something is nil.

Reply:
how about `(index uint32, val *Validator, ok bool)`? just don't see why we need an error here. validator is either present or not

Reply:
changed it to `ok bool`